### PR TITLE
Add default Huffman table if huffman table uninitialized

### PR DESCRIPTION
**Pull request by kim-mishra**
#### Description
Certain MJPEG frames do not include Huffman tables, this causes an `uninitialized Huffman table` error during `jpeg.Decode()`. These changes add a default Huffman Table in the case that one was not provide, so that the jpeg can be processed and transformed into an image. I tested these changes locally with cameras that provide and do not provide Huffman tables to ensure that the error is avoided, and for frames with initialized Huffman tables there is no change in behavior.

#### Reference issue
Fixes an uninitialized Huffman error that is returned from `jpeg.Decode()` when a jpeg Frame omits the Huffman Table

### CODE REVIEW

File: pkg/frame/compressed.go
Comment (edaniels) [inline pkg/frame/compressed.go:20]: can you return early if there is no error?
also, it would be better to use errors.As with https://cs.opensource.google/go/go/+/refs/tags/go1.20.3:src/image/jpeg/reader.go;l=18 to then check just the "uninitialized Huffman table" error so we can be a bit more specific

File: pkg/frame/compressed.go
Comment (edaniels) [inline pkg/frame/compressed.go:9]: licensing checks out 👍

File: pkg/frame/compressed.go
Comment (edaniels) [inline pkg/frame/compressed.go:29]: this should check that the split worked. otherwise a malicious input could crash the caller. this would be a good test

Comment (edaniels) [inline pkg/frame/compressed.go:29]: this is copying bytes in a potentially expensive way. can you make the slice in advance with the new size and then copy in order. it will also make these appends a tad easier to read!

File: pkg/frame/compressed.go
Comment (edaniels) [reply pkg/frame/compressed.go:20]: we do but we can eliminate the first part of that string to check

File: pkg/frame/compressed.go
Comment (kim-mishra) [reply pkg/frame/compressed.go:14]: This should work across devices in the case that the Huffman encoding info has been omitted for bandwidth optimization

Comment (edaniels) [reply pkg/frame/compressed.go:14]: Can you point where this comes from though?

Comment (kim-mishra) [reply pkg/frame/compressed.go:14]: The [JPEG wiki](https://en.wikipedia.org/wiki/JPEG) lists `dhtMarker` & `sosMarker` as dht & sos respectively.
This [RFC](https://www.rfc-editor.org/rfc/rfc2435) discusses a "Standard Huffman Table" and location of the citation is behind a paywall. This [thread](https://github.com/blackjack/webcam/issues/7) is where I got the table from. It works on cameras I have tested with and has been reported to work on other cameras as well.

Comment (at-wat) [reply pkg/frame/compressed.go:14]: > This [RFC](https://www.rfc-editor.org/rfc/rfc2435) discusses a "Standard Huffman Table" and location of the citation is behind a paywall.

https://www.w3.org/Graphics/JPEG/itu-t81.pdf
W3C seems putting it in public.

File: pkg/frame/compressed.go
Comment (at-wat) [reply pkg/frame/compressed.go:10]: Strictly speaking, when OSS licensed code is copied, we should include the full original license statements according to
```
The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.
```
in case of MIT.
https://github.com/filiptc/gorbit/blob/fa87ff39b68a6706306f34c318e0b9a5a3c97110/LICENSE

File: pkg/frame/compressed.go
Comment (kim-mishra) [inline pkg/frame/compressed.go:34]: @at-wat like this?

**Review by edaniels — CHANGES_REQUESTED** [review]
Done with another pass. Thanks for adding tests!

File: pkg/frame/compressed.go
Comment (kim-mishra) [reply pkg/frame/compressed.go:51]: Could that break other things? The function did not originally ever return nil

Comment (edaniels) [reply pkg/frame/compressed.go:52]: it would not

File: pkg/frame/compressed.go
Comment (edaniels) [inline pkg/frame/compressed.go:71]: so much better!

File: pkg/frame/compressed_test.go
Comment (edaniels) [inline pkg/frame/compressed_test.go:22]: all of these need the failnow too but you can just do Fatalf instead

**Review by edaniels — CHANGES_REQUESTED** [review]

File: pkg/frame/compressed_test.go
Comment (kim-mishra) [inline pkg/frame/compressed_test.go:51]: @edaniels like this?

Comment (edaniels) [reply pkg/frame/compressed_test.go:51]: yep!

**Review by edaniels — APPROVED** [review]
LGTM. Will wait on @at-wat for other approval

File: pkg/frame/compressed.go
Comment (at-wat) [reply pkg/frame/compressed.go:34]: Maybe better to move the copied part to separated file (if code inside `decodeMJPEG` and `addMotionDht` is also copied, current structure is fine)
Could you add a URL of the original repo as well?

Other things looks good to me!

Comment (kim-mishra) [reply pkg/frame/compressed.go]: Link is added! parts of the functions you mentioned are taken from the repo

**Review by at-wat — APPROVED** [review]
LGTM!

**Pull request merged**
